examples/advent_of_code/day13/knights_dinner_table.nit

```diff
@@ -0,0 +1,88 @@
+class Person
+	var name: String
+
+	var happiness_map = new HashMap[Person, Int]
+
+	fun compute_seatings(remaining: Set[Person]): Array[Array[Person]] do
+		if remaining.length == 1 then return [[self]]
+		var persons = new Array[Array[Person]]
+		var rem_set = remaining.clone
+		rem_set.remove self
+		for i in rem_set do
+			persons.add_all i.compute_seatings(rem_set)
+		end
+		for i in persons do i.unshift self
+		return persons
+	end
```

Comment:
Welp. Didn't work, either I'm using it wrong or there's a bug, in any case I'll leave the actual implementation for now